File: Subprogramas/Genericos/Generico.cs
//C Sharp (C#)

using System;

class Generico {
    public static T MaiorValor<T>(T a, T b) where T : IComparable<T> {
        return a.CompareTo(b) > 0 ? a : b;
    }

    static void Main() {
        Console.WriteLine("Maior entre 10 e 20: " + MaiorValor(10, 20));
        Console.WriteLine("Maior entre 3.14 e 2.71: " + MaiorValor(3.14, 2.71));
    }
}

File: Subprogramas/Genericos/generico.rs
//Rust 

fn maior_valor<T: PartialOrd>(a: T, b: T) -> T {
    if a > b { a } else { b }
}

fn main() {
    println!("Maior entre 10 e 20: {}", maior_valor(10, 20));
    println!("Maior entre 3.14 e 2.71: {}", maior_valor(3.14, 2.71));
}
